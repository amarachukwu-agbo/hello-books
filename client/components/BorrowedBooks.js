import React from 'react';
import { Link } from 'react-router-dom';

const BorrowedBooks = (props) => {
  const {
    books, returnBook, user, isSendingRequest, returnRequests,
  } = props;

  if (!books.length) {
    return (
        <div className="row center">
            <p className="grey-text">You have no borrowed books </p>
        </div>
    );
  }

  return (
        <div className="row">
            <table className="striped responsive-table">
                <thead>
                    <tr>
                        <th>Book</th>
                        <th>Date Borrowed</th>
                        <th>Status</th>
                        <th>Date Returned</th>
                    </tr>
                </thead>
                <tbody>
                    { books.map((book, index) =>
                        <tr key={ index }>
                            <td>{ book.borrowedBooks.title }</td>
                            <td> { book.createdAt.split('T')[0] } </td>
                            <td> { book.status } </td>
                            <td> { book.status === 'Returned' ? book.updatedAt.split('T')[0] : '' } </td>
                            { book.status !== 'Returned' &&
                            !returnRequests.find(request =>
                                request.bookId === book.bookId) &&
                            <td><button onClick = { returnBook.bind(null, user.id, book.bookId) }
                            className = "btn btn-small btn-wave waves-effect red"
                            disabled = { isSendingRequest } >Return</button></td>
                            }
                        </tr>)}
                </tbody>
            </table>
        </div>
  );
};

export default BorrowedBooks;
