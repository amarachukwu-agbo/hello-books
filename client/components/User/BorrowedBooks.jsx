import React from 'react';
import propTypes from 'prop-types';
/**
 * @description stateless component for rendering books borrowed
 * by a user
 *
 * @param {object} props - properties of the component
 *
 * @returns {Node} - react node containing the BorrowedBooks component
 */
const BorrowedBooks = (props) => {
  const {
    books, returnBook, user, isSendingRequest, returnRequests,
  } = props;

  if (!books.length) {
    return (
      <div className="row center">
        <p className="grey-text">You have no borrowed books </p>
      </div>
    );
  }

  return (
    <div className="section">
      <table className="striped responsive-table">
        <thead>
          <tr>
            <th>Book</th>
            <th>Date Borrowed</th>
            <th>Status</th>
            <th>Date Returned</th>
          </tr>
        </thead>
        <tbody>
          {books.map(book =>
            <tr key={book.id}>
              <td>{book.borrowedBooks.title}</td>
              <td> {book.createdAt.split('T')[0]} </td>
              <td> {book.status} </td>
              <td> {book.status === 'Returned' ?
                book.updatedAt.split('T')[0] : ''} </td>
              {book.status !== 'Returned' &&
                !returnRequests.find(request =>
                  request.bookId === book.bookId) &&
                <td><button onClick={() => returnBook(user.id, book.bookId)}
                  className="btn btn-small btn-wave waves-effect primary-button"
                  disabled={isSendingRequest} >Return</button></td>
              }
            </tr>)}
        </tbody>
      </table>
    </div>
  );
};

// Prop type validation
BorrowedBooks.propTypes = {
  books: propTypes.array.isRequired,
  returnBook: propTypes.func.isRequired,
  user: propTypes.object.isRequired,
  isSendingRequest: propTypes.bool,
  returnRequests: propTypes.array.isRequired,
};

export default BorrowedBooks;
