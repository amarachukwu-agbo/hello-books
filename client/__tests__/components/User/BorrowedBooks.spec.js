import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import mockData from '../../__mocks__/mockData';
import BorrowedBooks from '../../../components/User/BorrowedBooks';

const { users } = mockData;
const props = {
  returnBook: sinon.spy(() => Promise.resolve()),
  user: {
    id: 3,
  },
  isSendingRequest: false,
  returnRequests: users.profileResponse.user.userReturnRequests,
  books: users.profileResponse.user.userBooks,
};
describe('<BorrowedBooks />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<BorrowedBooks {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });
  it('calls returnBook function when button is clicked', () => {
    const wrapper = shallow(<BorrowedBooks {...props}/>);
    const returnButton = wrapper.find('#return-button7');
    returnButton.simulate('click');
    expect(props.returnBook.calledOnce).toEqual(true);
    expect(props.returnBook.calledWith(3, 67));
  });
  it('renders p if there are no books', () => {
    props.books = [];
    const wrapper = shallow(<BorrowedBooks {...props}/>);
    expect(wrapper.find('p').text())
      .toEqual('You have no borrowed books');
  });
});
