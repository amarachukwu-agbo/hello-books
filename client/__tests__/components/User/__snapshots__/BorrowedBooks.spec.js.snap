// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BorrowedBooks /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BorrowedBooks
    books={
      Array [
        Object {
          "bookId": 99,
          "borrowedBooks": Object {
            "author": "Andrew Hart",
            "title": "Lies That Bind Us",
          },
          "createdAt": "2018-06-24T01:06:38.794Z",
          "id": 4,
          "status": "Not Returned",
          "updatedAt": "2018-06-24T01:20:19.257Z",
          "userId": 3,
        },
        Object {
          "bookId": 67,
          "borrowedBooks": Object {
            "author": "Andrew Hart",
            "title": "New Book",
          },
          "createdAt": "2018-06-24T01:06:38.794Z",
          "id": 7,
          "status": "Not Returned",
          "updatedAt": "2018-06-24T01:20:19.257Z",
          "userId": 3,
        },
      ]
    }
    isSendingRequest={false}
    returnBook={[Function]}
    returnRequests={
      Array [
        Object {
          "bookId": 99,
          "comments": null,
          "createdAt": "2018-06-24T01:14:32.115Z",
          "id": 4,
          "returnRequests": Object {
            "author": "Andrew Hart",
            "title": "Lies That Bind Us",
          },
          "status": "Accepted",
          "updatedAt": "2018-06-24T01:20:19.244Z",
          "userId": 3,
        },
      ]
    }
    user={
      Object {
        "id": 3,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <table
        className="striped responsive-table"
      >
        <thead>
          <tr>
            <th>
              Book
            </th>
            <th>
              Date Borrowed
            </th>
            <th>
              Status
            </th>
            <th>
              Date Returned
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Lies That Bind Us
            </td>
            <td>
               
              2018-06-24
               
            </td>
            <td>
               
              Not Returned
               
            </td>
            <td>
               
              
               
            </td>
          </tr>
          <tr>
            <td>
              New Book
            </td>
            <td>
               
              2018-06-24
               
            </td>
            <td>
               
              Not Returned
               
            </td>
            <td>
               
              
               
            </td>
            <td>
              <button
                className="btn btn-small btn-wave waves-effect primary-button"
                disabled={false}
                id="return-button7"
                onClick={[Function]}
              >
                Return
              </button>
            </td>
          </tr>
        </tbody>
      </table>,
      "className": "section",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead>
            <tr>
              <th>
                Book
              </th>
              <th>
                Date Borrowed
              </th>
              <th>
                Status
              </th>
              <th>
                Date Returned
              </th>
            </tr>
          </thead>,
          <tbody>
            <tr>
              <td>
                Lies That Bind Us
              </td>
              <td>
                 
                2018-06-24
                 
              </td>
              <td>
                 
                Not Returned
                 
              </td>
              <td>
                 
                
                 
              </td>
            </tr>
            <tr>
              <td>
                New Book
              </td>
              <td>
                 
                2018-06-24
                 
              </td>
              <td>
                 
                Not Returned
                 
              </td>
              <td>
                 
                
                 
              </td>
              <td>
                <button
                  className="btn btn-small btn-wave waves-effect primary-button"
                  disabled={false}
                  id="return-button7"
                  onClick={[Function]}
                >
                  Return
                </button>
              </td>
            </tr>
          </tbody>,
        ],
        "className": "striped responsive-table",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <th>
                Book
              </th>
              <th>
                Date Borrowed
              </th>
              <th>
                Status
              </th>
              <th>
                Date Returned
              </th>
            </tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th>
                  Book
                </th>,
                <th>
                  Date Borrowed
                </th>,
                <th>
                  Status
                </th>,
                <th>
                  Date Returned
                </th>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Book",
                },
                "ref": null,
                "rendered": "Book",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Date Borrowed",
                },
                "ref": null,
                "rendered": "Date Borrowed",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Status",
                },
                "ref": null,
                "rendered": "Status",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Date Returned",
                },
                "ref": null,
                "rendered": "Date Returned",
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <tr>
                <td>
                  Lies That Bind Us
                </td>
                <td>
                   
                  2018-06-24
                   
                </td>
                <td>
                   
                  Not Returned
                   
                </td>
                <td>
                   
                  
                   
                </td>
              </tr>,
              <tr>
                <td>
                  New Book
                </td>
                <td>
                   
                  2018-06-24
                   
                </td>
                <td>
                   
                  Not Returned
                   
                </td>
                <td>
                   
                  
                   
                </td>
                <td>
                  <button
                    className="btn btn-small btn-wave waves-effect primary-button"
                    disabled={false}
                    id="return-button7"
                    onClick={[Function]}
                  >
                    Return
                  </button>
                </td>
              </tr>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "4",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td>
                    Lies That Bind Us
                  </td>,
                  <td>
                     
                    2018-06-24
                     
                  </td>,
                  <td>
                     
                    Not Returned
                     
                  </td>,
                  <td>
                     
                    
                     
                  </td>,
                  false,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Lies That Bind Us",
                  },
                  "ref": null,
                  "rendered": "Lies That Bind Us",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "2018-06-24",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "2018-06-24",
                    " ",
                  ],
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "Not Returned",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "Not Returned",
                    " ",
                  ],
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "",
                    " ",
                  ],
                  "type": "td",
                },
                false,
              ],
              "type": "tr",
            },
            Object {
              "instance": null,
              "key": "7",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td>
                    New Book
                  </td>,
                  <td>
                     
                    2018-06-24
                     
                  </td>,
                  <td>
                     
                    Not Returned
                     
                  </td>,
                  <td>
                     
                    
                     
                  </td>,
                  <td>
                    <button
                      className="btn btn-small btn-wave waves-effect primary-button"
                      disabled={false}
                      id="return-button7"
                      onClick={[Function]}
                    >
                      Return
                    </button>
                  </td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "New Book",
                  },
                  "ref": null,
                  "rendered": "New Book",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "2018-06-24",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "2018-06-24",
                    " ",
                  ],
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "Not Returned",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "Not Returned",
                    " ",
                  ],
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "",
                    " ",
                  ],
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <button
                      className="btn btn-small btn-wave waves-effect primary-button"
                      disabled={false}
                      id="return-button7"
                      onClick={[Function]}
                    >
                      Return
                    </button>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Return",
                      "className": "btn btn-small btn-wave waves-effect primary-button",
                      "disabled": false,
                      "id": "return-button7",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Return",
                    "type": "button",
                  },
                  "type": "td",
                },
              ],
              "type": "tr",
            },
          ],
          "type": "tbody",
        },
      ],
      "type": "table",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <table
          className="striped responsive-table"
        >
          <thead>
            <tr>
              <th>
                Book
              </th>
              <th>
                Date Borrowed
              </th>
              <th>
                Status
              </th>
              <th>
                Date Returned
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                Lies That Bind Us
              </td>
              <td>
                 
                2018-06-24
                 
              </td>
              <td>
                 
                Not Returned
                 
              </td>
              <td>
                 
                
                 
              </td>
            </tr>
            <tr>
              <td>
                New Book
              </td>
              <td>
                 
                2018-06-24
                 
              </td>
              <td>
                 
                Not Returned
                 
              </td>
              <td>
                 
                
                 
              </td>
              <td>
                <button
                  className="btn btn-small btn-wave waves-effect primary-button"
                  disabled={false}
                  id="return-button7"
                  onClick={[Function]}
                >
                  Return
                </button>
              </td>
            </tr>
          </tbody>
        </table>,
        "className": "section",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                <th>
                  Book
                </th>
                <th>
                  Date Borrowed
                </th>
                <th>
                  Status
                </th>
                <th>
                  Date Returned
                </th>
              </tr>
            </thead>,
            <tbody>
              <tr>
                <td>
                  Lies That Bind Us
                </td>
                <td>
                   
                  2018-06-24
                   
                </td>
                <td>
                   
                  Not Returned
                   
                </td>
                <td>
                   
                  
                   
                </td>
              </tr>
              <tr>
                <td>
                  New Book
                </td>
                <td>
                   
                  2018-06-24
                   
                </td>
                <td>
                   
                  Not Returned
                   
                </td>
                <td>
                   
                  
                   
                </td>
                <td>
                  <button
                    className="btn btn-small btn-wave waves-effect primary-button"
                    disabled={false}
                    id="return-button7"
                    onClick={[Function]}
                  >
                    Return
                  </button>
                </td>
              </tr>
            </tbody>,
          ],
          "className": "striped responsive-table",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th>
                  Book
                </th>
                <th>
                  Date Borrowed
                </th>
                <th>
                  Status
                </th>
                <th>
                  Date Returned
                </th>
              </tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th>
                    Book
                  </th>,
                  <th>
                    Date Borrowed
                  </th>,
                  <th>
                    Status
                  </th>,
                  <th>
                    Date Returned
                  </th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Book",
                  },
                  "ref": null,
                  "rendered": "Book",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Date Borrowed",
                  },
                  "ref": null,
                  "rendered": "Date Borrowed",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Status",
                  },
                  "ref": null,
                  "rendered": "Status",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Date Returned",
                  },
                  "ref": null,
                  "rendered": "Date Returned",
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <tr>
                  <td>
                    Lies That Bind Us
                  </td>
                  <td>
                     
                    2018-06-24
                     
                  </td>
                  <td>
                     
                    Not Returned
                     
                  </td>
                  <td>
                     
                    
                     
                  </td>
                </tr>,
                <tr>
                  <td>
                    New Book
                  </td>
                  <td>
                     
                    2018-06-24
                     
                  </td>
                  <td>
                     
                    Not Returned
                     
                  </td>
                  <td>
                     
                    
                     
                  </td>
                  <td>
                    <button
                      className="btn btn-small btn-wave waves-effect primary-button"
                      disabled={false}
                      id="return-button7"
                      onClick={[Function]}
                    >
                      Return
                    </button>
                  </td>
                </tr>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "4",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <td>
                      Lies That Bind Us
                    </td>,
                    <td>
                       
                      2018-06-24
                       
                    </td>,
                    <td>
                       
                      Not Returned
                       
                    </td>,
                    <td>
                       
                      
                       
                    </td>,
                    false,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Lies That Bind Us",
                    },
                    "ref": null,
                    "rendered": "Lies That Bind Us",
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        "2018-06-24",
                        " ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "2018-06-24",
                      " ",
                    ],
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        "Not Returned",
                        " ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "Not Returned",
                      " ",
                    ],
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        "",
                        " ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "",
                      " ",
                    ],
                    "type": "td",
                  },
                  false,
                ],
                "type": "tr",
              },
              Object {
                "instance": null,
                "key": "7",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <td>
                      New Book
                    </td>,
                    <td>
                       
                      2018-06-24
                       
                    </td>,
                    <td>
                       
                      Not Returned
                       
                    </td>,
                    <td>
                       
                      
                       
                    </td>,
                    <td>
                      <button
                        className="btn btn-small btn-wave waves-effect primary-button"
                        disabled={false}
                        id="return-button7"
                        onClick={[Function]}
                      >
                        Return
                      </button>
                    </td>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "New Book",
                    },
                    "ref": null,
                    "rendered": "New Book",
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        "2018-06-24",
                        " ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "2018-06-24",
                      " ",
                    ],
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        "Not Returned",
                        " ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "Not Returned",
                      " ",
                    ],
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        "",
                        " ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "",
                      " ",
                    ],
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <button
                        className="btn btn-small btn-wave waves-effect primary-button"
                        disabled={false}
                        id="return-button7"
                        onClick={[Function]}
                      >
                        Return
                      </button>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Return",
                        "className": "btn btn-small btn-wave waves-effect primary-button",
                        "disabled": false,
                        "id": "return-button7",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "Return",
                      "type": "button",
                    },
                    "type": "td",
                  },
                ],
                "type": "tr",
              },
            ],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
